forest/jasmine/traj2stats.py

```diff
@@ -140,175 +147,644 @@ def get_nearby_locations(traj: np.ndarray) -> Tuple[dict, dict, dict]:
+        no_windows = (end_stamp - start_stamp) // window
+    else:
+        # find starting and ending time
+        sys.stdout.write("Calculating the daily summary stats...\n")
+        time_list = stamp2datetime(traj[0, 3], tz_str)
+        time_list[3:6] = [0, 0, 0]
+        start_stamp = datetime2stamp(time_list, tz_str)
+        time_list = stamp2datetime(traj[-1, 6], tz_str)
+        time_list[3:6] = [0, 0, 0]
+        end_stamp = datetime2stamp(time_list, tz_str) + 3600 * 24
+        # if it starts from 2019-3-8 11 o'clock,
+        # then our daily summary starts from 2019-3-9)
+        window = 60 * 60 * 24
+        no_windows = (end_stamp - start_stamp) // window
+        if split_day_night:
+            no_windows *= 2
+
+    if no_windows <= 0:
+        raise ValueError("start time and end time are not correct")
+
+    for i in range(no_windows):
+        if split_day_night:
+            i2 = i // 2
+        else:
+            i2 = i
+        start_time = start_stamp + i2 * window
+        end_time = start_stamp + (i2 + 1) * window
+        start_time2 = 0
+        end_time2 = 0
+
+        current_time_list = stamp2datetime(start_time, tz_str)
+        year, month, day, hour = current_time_list[:4]
+        # take a subset, the starting point of the last traj <end_time
+        # and the ending point of the first traj >start_time
+        index_rows = (traj[:, 3] < end_time) * (traj[:, 6] > start_time)
+
+        stop1 = 0
+        stop2 = 0
+        if split_day_night:
+            current_time_list2 = current_time_list.copy()
+            current_time_list3 = current_time_list.copy()
+            current_time_list2[3] = 8
+            current_time_list3[3] = 20
+            start_time2 = datetime2stamp(current_time_list2, tz_str)
+            end_time2 = datetime2stamp(current_time_list3, tz_str)
+            if i % 2 == 0:
+                # datetime
+                index_rows = (
+                    (traj[:, 3] <= end_time2)
+                    * (traj[:, 6] >= start_time2)
+                    )
             else:
-                av_f_len = 0
-                sd_f_len = 0
-                av_f_dur = 0
-                sd_f_dur = 0
-            if len(pause_t_vec)>0:
-                av_p_dur = np.mean(pause_t_vec)
-                sd_p_dur = np.std(pause_t_vec)
+                # nighttime
+                index1 = (
+                    (traj[:, 6] < start_time2)
+                    * (traj[:, 3] < end_time)
+                    * (traj[:, 6] > start_time)
+                )
+                index2 = (
+                    (traj[:, 3] > end_time2)
+                    * (traj[:, 3] < end_time)
+                    * (traj[:, 6] > start_time)
+                )
+                stop1 = sum(index1) - 1
+                stop2 = sum(index1)
+                index_rows = index1 + index2
+
+        if sum(index_rows) == 0 and split_day_night:
+            res = [year, month, day] + [0 for _ in range(18)]
```

Comment:
`res = [year, month, day] + [0] * 18`? Also, it'd be great to explain what 18 is.